src/learner.cc

```diff
@@ -287,6 +316,8 @@ class LearnerImpl : public Learner {
             cfg_["predictor"] = "cpu_predictor";
             kv.second = "cpu_predictor";
             LOG(INFO) << "Switch gpu_predictor to cpu_predictor.";
+          } else {
```

Comment:
Probably I'm just missing it, but why not `else if (is_gpu_predictor)` so it doesn't trigger in the cpu_predictor case?